caffe2/CMakeLists.txt

```diff
@@ -177,7 +177,7 @@ if(BUILD_CAFFE2)
 endif()
 
 # Compile exposed libraries.
-add_library(caffe2 ${Caffe2_CPU_SRCS})
+add_library(caffe2 ${Caffe2_CPU_SRCS} $<TARGET_OBJECTS:c10>)
```

Comment:
Maybe just do

```
list(APPEND Caffe2_CPU_SRCS $<TARGET_OBJECTS:c10>)
```
above right after c10 directory is included? That way if you ultimately need to do Caffe2_GPU_SRCS, etc. you can do it as well.